### PR TITLE
DEVPROD-5289: Add ParsleySettings field to DBUser struct

### PR DESCRIPTION
**Pull request by minnakt**
DEVPROD-5289

PR is not actually as big as it says, 458 added lines were autogenerated.

### Description

This PR adds a new `ParsleySettings` field to the `DBUser` struct. In the future, `ParsleySettings` will hold more fields relating to user's preferences when viewing logs.

I added the `ParsleySettings.SectionsEnabled` field which describes if users should see sections in Parsley for Evergreen task logs. It should be enabled by default to expose users to the new feature.

### Testing
- Added GraphQL tests
- Added unit tests

### CODE REVIEW

File: rest/model/parsley.go [hunk 1]
```diff
@@ -0,0 +1,49 @@
+package model
+
+import (
+	"github.com/evergreen-ci/evergreen/model/parsley"
+	"github.com/evergreen-ci/utility"
+)
+
+type APIParsleySettings struct {
+	// Whether to render task logs with sections.
```
Comment (hadjri) [inline rest/model/parsley.go:9]: Nit: start comment with the name of the field

File: rest/model/parsley.go [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+}
+
+func (s *APIParsleySettings) ToService() parsley.Settings {
+	return parsley.Settings{
+		SectionsEnabled: s.SectionsEnabled,
+	}
+}
+
+type APIParsleyFilter struct {
+	// Regex to match in Parsley.
```
Comment (hadjri) [inline rest/model/parsley.go:29]: Same comment here

File: model/user/user_test.go [hunk 3]
```diff
@@ -821,6 +822,27 @@ func TestViewableProjectSettings(t *testing.T) {
...
+	newSettings := parsley.Settings{
+		SectionsEnabled: utility.FalsePtr(),
+	}
+	err := usr.UpdateParsleySettings(newSettings)
+	assert.NoError(t, err)
+	assert.Equal(t, utility.FromBoolPtr(usr.ParsleySettings.SectionsEnabled), false)
+
+	dbUser, err := FindOneById(usr.Id)
+	assert.NoError(t, err)
+	assert.NotNil(t, dbUser)
```
Comment (hadjri) [inline model/user/user_test.go:842]: This should be require.NotNil

**Review by hadjri — CHANGES_REQUESTED** [review]

**Comment by minnakt** [comment]
Just to add on to what @SupaJoon said, another factor worth mentioning is debuggability. If we were able to look up a user's specific Parsley settings, we could have troubleshooted [EVG-19872](https://jira.mongodb.org/browse/EVG-19872) and [DEVPROD-5058](https://jira.mongodb.org/browse/DEVPROD-5058) without the need of asking the user directly in each case. I think that New Relic might not be quite sufficient for that.

**Comment by khelif96** [comment]
That all sounds great to me @SupaJoon  @minnakt! Thank you for walking me through your reasoning and for the explanation!

File: model/user/user_test.go (hunk 3 above)
Comment (minnakt) [reply model/user/user_test.go]: ty!

File: graphql/schema/types/user.graphql [hunk 4]
```diff
@@ -80,6 +88,7 @@ type User {
   displayName: String!
   emailAddress: String!
   parsleyFilters: [ParsleyFilter!]!
+  parsleySettings: ParsleySettings
```
[the remark this answers is not in the document]
Comment (minnakt) [reply graphql/schema/types/user.graphql]: good point, done!

**Review by hadjri — APPROVED** [review]

**Comment by minnakt** [comment]
merging — failing project settings test should be unrelated 🤔

**Pull request merged**